Gems/AtomLyIntegration/CommonFeatures/Code/Source/Material/EditorMaterialComponentInspector.cpp

```diff
@@ -685,46 +705,37 @@ namespace AZ
                 return true;
             }
 
-            bool MaterialPropertyInspector::HasMaterialSource() const
-            {
-                return IsLoaded() && !m_editData.m_materialSourcePath.empty() &&
-                    AZ::StringFunc::Path::IsExtension(m_editData.m_materialSourcePath.c_str(), AZ::RPI::MaterialSourceData::Extension);
-            }
-
-            bool MaterialPropertyInspector::HasMaterialParentSource() const
-            {
-                return IsLoaded() && !m_editData.m_materialParentSourcePath.empty() &&
-                    AZ::StringFunc::Path::IsExtension(
-                        m_editData.m_materialParentSourcePath.c_str(), AZ::RPI::MaterialSourceData::Extension);
-            }
-
-            void MaterialPropertyInspector::OpenMaterialSourceInEditor() const
+            void MaterialPropertyInspector::OpenMenu()
             {
-                if (HasMaterialSource())
+                if (!IsLoaded())
                 {
-                    EditorMaterialSystemComponentRequestBus::Broadcast(
-                        &EditorMaterialSystemComponentRequestBus::Events::OpenMaterialEditor, m_editData.m_materialSourcePath);
+                    return;
                 }
-            }
 
-            void MaterialPropertyInspector::OpenMaterialParentSourceInEditor() const
-            {
-                if (HasMaterialParentSource())
+                QAction* action = nullptr;
+
+                QMenu menu(this);
+
+                action = menu.addAction(tr("Save As..."), [this] {
+                    const auto& defaultPath = AtomToolsFramework::GetUniqueFilePath(AZStd::string::format(
+                        "%s/Assets/untitled.%s", AZ::Utils::GetProjectPath().c_str(), AZ::RPI::MaterialSourceData::Extension));
+                    SaveMaterial(defaultPath);
+                });
+
+                if (IsSourceMaterial(m_editData.m_materialSourcePath))
                 {
-                    EditorMaterialSystemComponentRequestBus::Broadcast(
-                        &EditorMaterialSystemComponentRequestBus::Events::OpenMaterialEditor, m_editData.m_materialParentSourcePath);
+                    const auto& materialSourceRelativePath = GetRelativePath(m_editData.m_materialSourcePath);
+                    action = menu.addAction(tr("Save Over \"%1\"...").arg(materialSourceRelativePath.c_str()), [this] {
+                        SaveMaterial(m_editData.m_materialSourcePath);
```

Comment:
Did you make sure this works okay with really long paths?

Reply:
If it's a really long path, it will be a really long menu entry. We might be able to use eliding or cut it off it's too long.

Reply:
What is the next step of "Save over ..." do? Do we show a dialog for user to confirm the saving behavior, since there is a "..."? If yes, can we only show the material file name, instead of the whole file path? this would be a bad UI layout if the file menu is really long. 

Also, I see that you also include the file path on the dialog, it will trigger the same issue when the file path is long. I suggest we should do the following:
1. Show only the file name
2. User can mouse over the file name to see the full file path in the tool tip.
3. Behind the file name, add an "open file" icon for user to click and open the file in the material editor.

Reply:
The ... Is because it's still opening a file dialog for the user to confirm, cancel, or select a different file.
Yes, I will switch them all to only show the filename.
Which icon would you like to be used? I will see if it is possible to embed it in the label text or if I have to make more extensive changes.